src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/VisibleMemberTable.java

```diff
@@ -106,8 +107,38 @@
         ANNOTATION_TYPE_MEMBER_REQUIRED,
         PROPERTIES;
 
-        public static final EnumSet<Kind> summarySet = EnumSet.range(INNER_CLASSES, METHODS);
-        public static final EnumSet<Kind> detailSet = EnumSet.range(ENUM_CONSTANTS, METHODS);
+        private static final EnumSet<Kind> defaultSummarySet = EnumSet.of(
+                INNER_CLASSES, FIELDS, CONSTRUCTORS, METHODS);
+        private static final EnumSet<Kind> enumSummarySet = EnumSet.of(
+                INNER_CLASSES, ENUM_CONSTANTS, FIELDS, METHODS);
+        private static final EnumSet<Kind> annotationSummarySet = EnumSet.of(
+                FIELDS, ANNOTATION_TYPE_MEMBER_OPTIONAL, ANNOTATION_TYPE_MEMBER_REQUIRED);
+        private static final EnumSet<Kind> defaultDetailSet = EnumSet.of(
+                FIELDS, CONSTRUCTORS, METHODS);
+        private static final EnumSet<Kind> enumDetailSet = EnumSet.of(
+                ENUM_CONSTANTS, FIELDS, METHODS);
```

Comment:
I'm a bit surprised by this, if only because we've had problems with tables like this in the past.

Is it possible to have a single combined list where some of the entries will give effectively empty tables which can be suppressed?

Case in point for an issue: why does `enumSummarySet` have `INNER_CLASSES` but `annotationSummarySet` does not.